libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -211,6 +229,77 @@ void HarmonicNotchFilter<T>::expand_filter_count(uint16_t total_notches)
     delete[] _old_filters;
 }
 
+/*
+  set the center frequency of a single notch
+ */
+template <class T>
+void HarmonicNotchFilter<T>::set_center_frequency(uint16_t idx, float notch_center, float spread_mul, uint8_t harmonic_mul)
+{
+    const float nyquist_limit = _sample_freq_hz * 0.48f;
+    auto &notch = _filters[idx];
+
+    // scale the notch with the harmonic multiplier
+    notch_center *= harmonic_mul;
+
+    /* disable the filter if its center frequency is above the nyquist
+       frequency
+
+       NOTE: should this be notch_center*spread_mul ? As it is here we
+       do sometimes run the upper notch in a double or triple notch at
+       higher than the nyquist.
+    */
+    if (notch_center >= nyquist_limit) {
+        notch.disable();
+        return;
+    }
+
+    // the minimum frequency for a harmonic is the base minimum
```

Comment:
The original code prevented the notch going below a frequency that was < bandwidth/2 - I don't see that here but can't tell whether it is covered or not. Funny things happen in the maths if it is true. Be good if your unit test specifically exercised these.

Reply:
that is in NotchFilter<T>::calculate_A_and_Q() and is still there

Reply:
Actually its slightly different. In that function its > bw*0.5 in this its bw*0.52. Since this function used to be used in preference I think we should change the other value to 0.52. Probably the same for the nyquist calculation of 0.48

Reply:
Actually your statement is correct but not sufficient. calculate_A_and_Q() is only called in init() it is not called continuously - in this function you need to compare actively with the new center not with your initial calculation. That's why its important to have the check here or add it to init_with_A_and_Q()

Reply:
in master it is also only called on init(). Presumably that is because it is only impacting the Q, so only needed on calculate_A_and_Q()?
That matches what is done with this PR. The init() call in HarmonicNotchFilter() limits to 0.52*bw

Reply:
I *think* you are correct so resolving this